framework/sandstone_p.h

```diff
@@ -277,13 +277,84 @@ struct SandstoneApplication : public test_the_test_data<SandstoneConfig::Debug>
     };
     static constexpr auto DefaultOutputFormat = SANDSTONE_DEFAULT_LOGGING;
 
-    enum ForkMode : int8_t {
-        no_fork,
-        fork_each_test,
-        exec_each_test,
-        child_exec_each_test,       // when parent is exec_each_test
+    // test execution
+    bool selftest = false;
+    bool ud_on_failure = false;
+    bool use_strict_runtime = false;
+
+    // logging parameters
+    int verbosity = -1;
+    int max_messages_per_thread = 5;
+    unsigned max_logdata_per_thread = 128;
+    OutputFormat output_format = DefaultOutputFormat;
+    uint8_t output_yaml_indent = 0;
+    bool log_test_knobs = false;
+};
+
+struct SandstoneApplication : public test_the_test_data<SandstoneConfig::Debug>
+{
+    // Isolate variables set during opts parsing into a separate struct.
+    // This struct is passed to parse_cmdline(), instead of sApp.
+    struct Cfg {
+        enum class ForkMode : int8_t {
+            no_fork,
+            fork_each_test,
+            exec_each_test,
+            child_exec_each_test,       // when parent is exec_each_test
+        };
+
+        int requested_quality = DefaultQualityLevel;
+        std::string file_log_path;
+        const char *syslog_ident = nullptr;
+
+        MonotonicTimePoint starttime = MonotonicTimePoint::clock::now();
+        MonotonicTimePoint endtime;
+        ShortDuration test_time = {};
+        ShortDuration max_test_time = {};
+        ShortDuration timeout_to_kill = std::chrono::seconds(20);
+        ShortDuration delay_between_tests = std::chrono::milliseconds(5);
+
+#ifdef NDEBUG
+        static constexpr struct {
+            size_t size() const { return 0; }
+            char *c_str() const { return nullptr; }
+        } gdb_server_comm = {};
+#else
+        std::string gdb_server_comm;
+#endif
+
+        bool fatal_skips = false;
+
+        ForkMode fork_mode =
+#ifdef _WIN32
+            ForkMode::exec_each_test;
+#else
+            ForkMode::fork_each_test;
+#endif
+
+        bool ignore_mce_errors = false;
+        bool ignore_os_errors = false;
+        bool force_test_time = false;
+        bool service_background_scan = false;
+        bool vary_frequency_mode = false;
+        bool vary_uncore_frequency_mode = false;
+        bool include_optional = false;
+        int inject_idle = 0;
+
+        int retest_count = 10;
+        int total_retest_count = -2;
+        int max_test_count = INT_MAX;
+        int max_test_loop_count = 0;
+
+        int thermal_throttle_temp = DefaultTemperatureThreshold;
+
+        int thread_count;
+
+        std::unique_ptr<DeviceScheduler> device_scheduler = nullptr;
     };
 
+    Cfg cfg;
+
```

Comment:
Recommendation: insert here:
```c++
    using OutputFormat = ShmemCfg::OutputFormat; // or TestConfiguration::OutputFormat
```

To retain compatibility with `SandstoneApplication::OutputFormat::whatever` uses.